legend-engine-xt-relationalStore-pure/src/main/resources/core_relational/relational/pureToSQLQuery/pureToSQLQuery.pure

```diff
@@ -2383,9 +2383,12 @@ function meta::relational::functions::pureToSqlQuery::processSlice(f:FunctionExp
    let mainQuery = processValueSpecification($f.parametersValues->at(0), $currentPropertyMapping, $operation, $vars, $state, $joinType, $nodeId, $aggFromMap, $context, $extensions)->toOne()->cast(@SelectWithCursor);
    let mainSelect = $mainQuery.select;
 
-   let fromRow = processValueSpecification($f.parametersValues->at(1), $currentPropertyMapping, $operation, $vars, $state, JoinType.LEFT_OUTER, $nodeId, $aggFromMap, $context, $extensions)->toOne()->cast(@SelectWithCursor).select.columns->at(0)->cast(@Literal);
-   let toRow = processValueSpecification($f.parametersValues->at(2), $currentPropertyMapping, $operation, $vars, $state, JoinType.LEFT_OUTER, $nodeId, $aggFromMap, $context, $extensions)->toOne()->cast(@SelectWithCursor).select.columns->at(0)->cast(@Literal);
-
+   let param1 = processValueSpecification($f.parametersValues->at(1), $currentPropertyMapping, $operation, $vars, $state, JoinType.LEFT_OUTER, $nodeId, $aggFromMap, $context, $extensions)->toOne()->cast(@SelectWithCursor).select.columns->at(0);
+   assert($param1->instanceOf(Literal),'Only literals are supported inside the slice function, you are passing a ' + $param1->type()->toString());
+   let fromRow = $param1->cast(@Literal);
+   let param2 = processValueSpecification($f.parametersValues->at(2), $currentPropertyMapping, $operation, $vars, $state, JoinType.LEFT_OUTER, $nodeId, $aggFromMap, $context, $extensions)->toOne()->cast(@SelectWithCursor).select.columns->at(0);
+   assert($param2->instanceOf(Literal),'Only literals are supported inside the slice function, you are passing a ' + $param2->type()->toString());
```

Comment:
Can we change the message to be more succinct.

Something like 'Invalid type for parameter 0. Expected=Lietral, Found=X'

Reply:
changed